client.go

```diff
@@ -1125,6 +1125,75 @@ func (c *Client) GetClient() *http.Client {
 	return c.httpClient
 }
 
+// Clone returns a clone of the original client.
+//
+// Be carefull when using this function:
+// - Interface values are not deeply cloned. Thus, both the original and the clone will use the
+// same value.
+// - This function is not safe for concurrent use. You should only use this when you are sure that
+// the client is not being used by any other goroutine.
+//
+// Since v2.12.0
+func (c *Client) Clone() *Client {
```

Comment:
@fabiante Thanks for updating the PR. I believe I requested this method replacement with -
```go
func (c *Client) Clone() *Client {
	// dereference the pointer and copy the value
	cc := *c
	return &cc
}
```

I still see the previous code. Did you face any issues?

Reply:
Oh right. Id skipped over that part I guess. I will look into that.

Reply:
I originally intended to implement the copy function like this. I did **not** do it because the linter pointed out that the dereferencing would copy the mutex value: `assignment copies lock value`

With the fields changed to pointers, this warning is gone. But is this really okay?

AFAIK there will still be only one mutext which is now shared across clients. Is that the way we want to go? Fine by me but I think I don't yet have a deep understanding of how this could effect users.

I will push the change in case we are fine with that approach.

Reply:
@fabiante I will validate and get back

Reply:
Resolving since your latest comment adresses the same topic.